common4j/src/main/com/microsoft/identity/common/java/platform/AbstractKeyStoreKeyManager.java

```diff
@@ -0,0 +1,289 @@
+                                      @NonNull final String keyAlias,
+                                      @Nullable final KeyStore.PasswordProtection passwordProtection) throws KeyStoreException {
+        mKeyAlias = keyAlias;
+        mKeyStore = keyStore;
+        mPasswordProtection = passwordProtection;
+    }
+
+    @Override
+    public boolean exists() {
+        boolean exists = false;
+
+        try {
+            exists = mKeyStore.containsAlias(mKeyAlias);
+        } catch (final KeyStoreException e) {
+            Logger.error(
+                    TAG,
+                    "Error while querying KeyStore",
+                    e
+            );
+        }
+
+        return exists;
+
+    }
+
+    @Override
+    public boolean hasThumbprint(final byte[] thumbprint) {
+        try {
+            return Arrays.equals(thumbprint, getThumbprint());
+        } catch (final ClientException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public Date getCreationDate() throws ClientException {
+        try {
+            return mKeyStore.getCreationDate(mKeyAlias);
+        } catch (final KeyStoreException e) {
+            Logger.error(
+                    TAG,
+                    "Error while getting creation date for alias " + mKeyAlias,
+                    e
+            );
+            throw new ClientException(ClientException.KEYSTORE_NOT_INITIALIZED, e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean clear() {
+        boolean deleted = false;
+
+        try {
+            mKeyStore.deleteEntry(mKeyAlias);
+            deleted = true;
+        } catch (final KeyStoreException e) {
+            Logger.error(
+                    TAG,
+                    "Error while clearing KeyStore",
+                    e
+            );
+        }
+
+        return deleted;
+    }
+
+
+    /**
+     * Retrieve the entry stored in this particular alias.
+     *
+     * @return the Entry in question, or null if the entry does not exist.
+     * @throws UnrecoverableEntryException
+     * @throws NoSuchAlgorithmException
+     * @throws KeyStoreException
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public K getEntry() throws UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException {
+        return (K) mKeyStore.getEntry(mKeyAlias, mPasswordProtection);
+    }
+
+    @Override
+    public void importKey(@NonNull final byte[] jwk, @NonNull final String algorithm) throws ClientException {
+        throw new UnsupportedOperationException("This is not currently supported");
+    }
+
+    @Override
+    public byte[] getThumbprint() throws ClientException {
+        final Exception exception;
+        final String errCode;
+
+        try {
+            final K entry = getEntry();
+
+            if (entry instanceof KeyStore.PrivateKeyEntry) {
+                return getRsaThumbprint((KeyStore.PrivateKeyEntry) entry).getBytes(UTF8);
```

Comment:
A Private Key is not necessarily an RSA key.

If we want to keep this here, then we probably want to validate that first.

Reply:
Addressed